### PR TITLE
[deliver] wait for candidate builds to appear if not available immediately

### PR DESCRIPTION
**Pull request by mjovkovic**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
In deliver action, when submitting apps automatically, sometimes (maybe in 20% of cases) candidate builds can not be fetched immediately after the binary upload. Submit for review fails with the message `Could not find any available candidate builds on App Store Connect to submit`.
https://github.com/fastlane/fastlane/issues/10411

Log excerpt from the failed submission:
```
[16:54:28]: Finished the upload to iTunes Connect
[16:54:28]: Selecting the latest build...
+---------------------+--------------------------------------+
| Lane Context |
+---------------------+--------------------------------------+
| DEFAULT_PLATFORM | ios |
| PLATFORM_NAME | ios |
| LANE_NAME | ios deploy |
| VERSION_NUMBER | ***** |
| BUILD_NUMBER | ***** |
| PRODUCE_APPLE_ID | ***** |
| CERT_FILE_PATH | ***** |
| CERT_CERTIFICATE_ID | ***** |
| SIGH_PROFILE_PATH | ***** |
| SIGH_PROFILE_PATHS | ***** |
| SIGH_UDID | *****|
| SIGH_UUID | ***** |
| SIGH_NAME | ***** |
| SIGH_PROFILE_TYPE | app-store |
| IPA_OUTPUT_PATH | ***** |
| DSYM_OUTPUT_PATH | ***** |
| XCODEBUILD_ARCHIVE | ***** |
+---------------------+--------------------------------------+
[16:54:29]: Could not find any available candidate builds on iTunes Connect to submit
```

### Description
I've added a wait loop that tries to fetch candidate builds every 30 seconds. It raises and error if candidate builds don't appear in 5 minutes.

I've used the changes on our CI machine, and it solved the issue. Here's the example output when builds are available:
```
[19:39:22]: Finished the upload to App Store Connect
[19:39:23]: Selecting the latest build...
[19:39:25]: Waiting App Store Connect processing for build ******... this might take a while...
```

Here's the example output when wait loop fixed the issue which used to cause the deliver to fail:
```
[18:51:22]: Finished the upload to App Store Connect
[18:51:22]: Selecting the latest build...
[18:51:23]: Waiting for candidate builds to appear...
[18:51:54]: Waiting App Store Connect processing for build ******... this might take a while...
```
Note the `Waiting for candidate builds to appear...` message indicating the wait for candidates loop is used.

### CODE REVIEW

**Comment by mjovkovic** [comment]
I signed it!

File: deliver/lib/deliver/submit_for_review.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ def wait_for_build(app)
       build = nil
 
       loop do
-        latest_build = find_build(app.latest_version.candidate_builds)
+        latest_build = find_build(app.latest_version)
```
Comment (joshdholtz) [inline deliver/lib/deliver/submit_for_review.rb]: `find_build` makes this have a loop in a loop which feels a bit too loopy 😊 Is it possible to reuse this same loop here?

Maybe we simply this and just do something like...
```rb
loop do
  # Waits for candidate builds to appear
  candidate_builds = app.latest_version.candidate_builds
  if (candidate_builds || []).count == 0
    UI.message("Waiting for candidate builds to appear...")
    unless (Time.now - start) > (60 * 5)
      next
    else
      UI.user_error!("Could not find any available candidate builds on App Store Connect to submit")
    end
  end
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Thanks for submitting this PR 😊 Left a comment with an idea of making this PR a bit smaller and more streamlined to work within the existing loop that we have. Lemme know your thoughts when you get a chance 🙃

File: deliver/lib/deliver/submit_for_review.rb (hunk 1 above)
Comment (mjovkovic) [reply deliver/lib/deliver/submit_for_review.rb]: @joshdholtz that totally makes sense, I'll update and test the code as soon as I can!

**Comment by mjovkovic** [comment]
@joshdholtz I've updated the code as requested, let me know what you think :)

File: deliver/lib/deliver/submit_for_review.rb [hunk 2]
```diff
@@ -58,8 +58,21 @@ def wait_for_build(app)
       build = nil
 
       loop do
-        latest_build = find_build(app.latest_version.candidate_builds)
+        # Sometimes candidate_builds don't appear immediately after submittion
+        # Wait for candidate_builds to appear on iTunes
```
Comment (joshdholtz) [inline deliver/lib/deliver/submit_for_review.rb]: Small nit pick but can we change `iTunes` to `App Store Connect` in this comment? 🙃

**Review by joshdholtz — CHANGES_REQUESTED** [review]
One small nit pick otherwise this is 💯 😊

File: deliver/lib/deliver/submit_for_review.rb (hunk 2 above)
Comment (mjovkovic) [reply deliver/lib/deliver/submit_for_review.rb]: @joshdholtz done! :)

**Review by joshdholtz — APPROVED** [review]
This is 🔥 Thanks for adding this ❤️

**Comment by peterfox1** [comment]
nice one @mjovkovic we've been getting this issue more often recently too

File: deliver/lib/deliver/submit_for_review.rb (hunk 1 above)
Comment (Ajaykn21) [reply deliver/lib/deliver/submit_for_review.rb]: @mjovkovic @joshdholtz 
This fix is for deliver only. 
We have seen few issues in pilot.
Can this be a fix for pilot as well ?

**Pull request merged**